metrics/web/docs/features/remote_config/01_remote_config_design.md

```diff
@@ -0,0 +1,73 @@
+
+For our purposes, we need to specify parameters for the `FPS Monitor`, `Login Form`, and `Renderer Display` features.
+
+Example of possible key-values are:
+
+```json
+isLoginFormEnabled: true
+isFpsMonitorEnabled: false
+isRendererDisplayEnabled: false
+```
+
+## Metrics application
+
+### Domain layer
+
+In the application domain layer, we should add an ability to read values from the `Remote Config`. For this purpose, we should:
+
+1. Create the `RemoteConfig` entity.
+2. Add the `getConfig` method to the `RemoteConfigRepository`.
+3. Implement the `GetRemoteConfigUseCase` class.
+
+So, the domain layer should look like this:
+
+![Domain layer diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/remote_config_design/metrics/web/docs/features/remote_config/diagrams/remote_config_domain_class.puml)
+
+### Data layer
+
+The `FirebaseRemoteConfigRepository` of the data layer should implement the methods from the `RemoteConfigRepository` interface. To do that, we should create a `FirebaseRemoteConfigAdapter` class.
+
+The following class diagram represents the classes of the data layer required for this feature:
+
+![Data layer diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/remote_config_design/metrics/web/docs/features/remote_config/diagrams/remote_config_data_class.puml)
+
+### Presentation layer
+
+Once we've created a domain and data layers, it's time to create a presentation layer. This layer contains the `RemoteConfigNotifier` - the class that manages `Remote Config` values.
+
+1. Create a `RemoteConfigNotifier` class.
+2. Implement the fields that stand for `Remote Config` values.
+3. Inject the notifier into the `InjectionContainer`.
+
+The following class diagram represents the classes of the presentation layer required for this feature:
+
+![Presentation layer diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/remote_config_design/metrics/web/docs/features/remote_config/diagrams/remote_config_presentation_class.puml)
+
+So, when the user enters the application, the following sequence diagram describes how the application applies `Remote Config` values:
+
+![Sequence diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/remote_config_design/metrics/web/docs/features/remote_config/diagrams/remote_config_sequence.puml)
+
+Let's consider the mechanism of applying the `Remote Config` values in the application.
+
+When the user enters the application, he sees the `LoadingPage` while waiting until the `_initializeRemoteConfig` method of the `RemoteConfigNotifier` finishes. After that, the `isLoading` status of the `RemoteConfigNotifier` sets to `false`, and the user can see the UI with `Remote Config` values applied.
```

Comment:
```suggestion
Let's consider the mechanism of applying the `Remote Config` values in the application. When a user enters the application, he or she stays on the `LoadingPage` until the `_initializeRemoteConfig` method of the `RemoteConfigNotifier` finishes. Once the initializing completes, the `isLoading` status of the `RemoteConfigNotifier` is set to `false`. The user then can proceed to the application with the configurations applied. 
```
We should keep in mind here that the described behavior requires changes to the `LoadingPage` implementation. Maybe, it would be even a separate ticket as this seems to be not a small task.